streaming-twitter/src/main/scala/org/apache/spark/streaming/twitter/TwitterInputDStream.scala

```diff
@@ -85,10 +85,8 @@ class TwitterReceiver(
         }
       })
 
-      val query = new FilterQuery
-      if (filters.size > 0) {
-        query.track(filters.mkString(","))
-        newTwitterStream.filter(query)
+      if (query.isDefined) {
+        newTwitterStream.filter(query.get)
       } else {
```

Comment:
It looks like the query option was being available as a seq of string, which then was converted to FilterQuery, and doing so we were hiding the implementation detail specific interface from the user. What were the limitations of using that strategy ?